### PR TITLE
Add support for user expiry

### PR DESCRIPTION
**Pull request by mattbostock**
After comments on #1402, adds support for user expiry so we can set a
user's expiry date when we learn their leaving date, rather than having
to push a release out on the day.

Credit goes to @philandstuff who originally raised this as #1487. I've added some validation for the date format.

### CODE REVIEW

File: modules/govuk/manifests/user.pp [hunk 1]
```diff
@@ -68,9 +74,12 @@
     $membership = 'minimum'
   }
 
+  validate_re($expiry, '^(absent|\d{4}-\d{2}-\d{2})$')
```
Comment (surminus) [inline modules/govuk/manifests/user.pp]: Not a biggie, but it's a shame Puppet doesn't seem to have a function to validate a date specifically. I am not sure what the behaviour would be if an invalid date is inputted (like 2016-02-30).

Comment (mattbostock) [inline modules/govuk/manifests/user.pp]: It'd be nice, but complex to implement ourselves and I don't think there's anything readily available that we could use.

**Comment by rjw1** [comment]
useful addition. merge when ready

**Comment by philandstuff** [comment]
:whale:

**Pull request merged**